chat/src/main/java/com/github/twitch4j/chat/events/channel/UserStateEvent.java

```diff
@@ -0,0 +1,131 @@
+package com.github.twitch4j.chat.events.channel;
+
+import com.github.twitch4j.chat.events.AbstractChannelEvent;
+import com.github.twitch4j.common.events.domain.EventChannel;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+
+/**
+ * This event gets called when a user joins a channel or sends a PRIVMSG to a channel.
+ */
+@Value
+@Getter
+@EqualsAndHashCode(callSuper = false)
+public class UserStateEvent extends AbstractChannelEvent {
+
+    /**
+     * Hexadecimal RGB color code, or null if it is never set.
+     */
+    private String color;
+
+    /**
+     * The user's display name, or null if it is never set.
+     */
+    private String displayName;
+
+    /**
+     * List of emote sets.
+     */
+    private List<String> emoteSets;
+
+    /**
+     * Badges map of tags to versions.
+     */
+    private Map<String, String> badges;
+
+    /**
+     * Metadata related to the chat badges.
+     */
+    private Map<String, String> badgeInfo;
+
+    /**
+     * Event constructor
+     *
+     * @param channel     The channel that this event originates from.
+     * @param displayName The display version of the user name
+     * @param color       Hexadecimal RGB color code of the user
+     * @param emoteSets   The emote sets the user is entitled to use.
+     * @param badges      Chat badges of the user and the version of each badge
+     * @param badgeInfo   Metadata related to the user chat badges
+     */
+    public UserStateEvent(EventChannel channel, String displayName, String color,
+                          List<String> emoteSets, Map<String, String> badges, Map<String, String> badgeInfo) {
+        super(channel);
+        this.displayName = displayName;
+        this.color = color;
+        this.emoteSets = Collections.unmodifiableList(emoteSets);
+        this.badges = Collections.unmodifiableMap(badges);
+        this.badgeInfo = Collections.unmodifiableMap(badgeInfo);
+    }
+
+    /**
+     * @return true if user is subscribed, false otherwise
+     */
+    public boolean isSubscriber() {
+        return this.badges.containsKey("subscriber");
```

Comment:
In addition to delegating to `IRCMessageEvent`, these badge lookups could be done via `clientPermissions` like so: `messageEvent.getClientPermissions().contains(CommandPermission.SUBSCRIBER)`

Reply:
Is that ok to add new items in `CommandPermission` enum and parse logic at `com.github.twitch4j.common.util.TwitchUtils#getPermissionsFromTags` for `global_mod` and `admin` badges?

Reply:
`global_mod` is no longer used by twitch so you can skip that one. `admin` I've never seen in the wild so it's up to you. To make this change you'd also need to update `TwitchUtils.getPermissionsFromTags`